zerver/views/thumbnail.py

```diff
@@ -39,7 +40,14 @@ def backend_serve_thumbnail(
         assert isinstance(maybe_user_profile, UserProfile)
         realm = maybe_user_profile.realm
 
-    if not validate_thumbnail_request(realm, maybe_user_profile, url):
+    is_authorized = validate_thumbnail_request(realm, maybe_user_profile, url)
+    if is_authorized is None:
+        if get_best_match(
+            request.META.get("HTTP_ACCEPT", ""),
+            ["image/jpeg", "image/png", "image/webp", "image/avif", "image/svg+xml"],
+        ):
+            return redirect(generate_thumbnail_url("static/images/errors/imageNotFound.png"))
```

Comment:
Oh, and tests are failing because this line (line 49) doesn't get run during tests.  Because you've not written a new test which gets to this line!

Reply:
@alexmv read through everything, I think I actually get what I need to do now :D thank you for the very helpful review!! it's kinda late in Astana rn, I will try to implement things you mentioned tomorrow. thanks again